fs_test.go

```diff
@@ -221,3 +221,25 @@ func TestIsEmpty(t *testing.T) {
 		}
 	}
 }
+
+func TestHasPrefix(t *testing.T) {
+	cases := []struct {
+		dir    string
+		prefix string
+		want   bool
+	}{
+		{"/usr/bin", "/", true},
+		{"/usr/bin", "/usr", true},
+		{"/usr/bin", "/usr/bin", true},
+		{"/usr/bin", "/var", false},
+		{"/usr/bin", "/usr/local/bin", false},
+		{"/usr/bin", "/usr/bin2", false},
```

Comment:
Let's make sure we have (at least) one test case for the situation where the only character in a path component is uncased - e.g. `foo/⌘/bar`.